packages/discord.js/src/util/Util.js

```diff
@@ -210,6 +235,51 @@ function escapeSpoiler(text) {
   return text.replaceAll('||', '\\|\\|');
 }
 
+/**
+ * Escapes escape characters in a string.
+ * @param {string} text Content to escape
+ * @returns {string}
+ */
+function escapeEscape(text) {
+  return text.replaceAll('\\', '\\\\');
+}
+
+/**
+ * Escapes heading characters in a string.
+ * @param {string} text Content to escape
+ * @returns {string}
+ */
+function escapeHeading(text) {
+  return text.replaceAll(/^( {0,2}[*-] +)?(#{1,3} )/gm, '$1\\$2');
+}
+
+/**
+ * Escapes bulleted list characters in a string.
+ * @param {string} text Content to escape
+ * @returns {string}
+ */
+function escapeBulletedList(text) {
+  return text.replaceAll(/^( *)[*-]( +)/gm, '$1\\-$2');
+}
+
+/**
+ * Escapes numbered list characters in a string.
+ * @param {string} text Content to escape
+ * @returns {string}
+ */
+function escapeNumberedList(text) {
+  return text.replaceAll(/^( *\d+)\./gm, '$1\\.');
+}
+
+/**
+ * Escapes masked link characters in a string.
+ * @param {string} text Content to escape
+ * @returns {string}
+ */
+function escapeMaskedLink(text) {
+  return text.replaceAll(/\[.+\]\(.+\)/gm, '\\$0');
+}
```

Comment:
There is no capturing group here, so this function effectively returns `$0` always.

Reply:
adding a test for all these functions is needed ig?